cou/steps/execute.py

```diff
@@ -36,13 +37,17 @@ async def _run_step(step: BaseStep, prompt: bool, overwrite_progress: bool = Fal
     in CLI output. True to overwrite and False (the default) to persist.
     :type overwrite_progress: bool
     """
-    if isinstance(step, UpgradeStep):
-        progress_indicator.start(step.description)
-        await step.run()
-        if not overwrite_progress:
-            progress_indicator.succeed()
-    else:
-        await step.run()
+    try:
+        if isinstance(step, UpgradeStep):
+            progress_indicator.start(step.description)
+            await step.run()
+            if not overwrite_progress:
+                progress_indicator.succeed()
+        else:
+            await step.run()
+    except HaltUpgradeExecution:
+        progress_indicator.fail()
+        step.sub_steps = []
```

Comment:
We should not change a sub-step of any step in the plan during execution, that's not right. You can cancel them, but I do not think that it's good to have sub-steps for `UnitUpgradeStep(description="Run the instance-count to upgrade", ...)`. I'm ok to handle it in execution, but I will expect that it works like `continue`.

Reply:
From our discussion:
```python
try:
    await apply_step(sub_step, prompt, overwrite_substeps_progress)
except COUBreakException:
    progress_indicator.failed()
    break
```